core/unit_test/TestArray.cpp

```diff
@@ -250,28 +250,28 @@ constexpr bool test_begin_end() {
   static_assert(begin(a1) == &a1[0]);
   static_assert(end(a1) == &a1[0] + a1.size());
 
-  Kokkos::Array<double, 0> n0{};
+  [[maybe_unused]] Kokkos::Array<double, 0> n0{};
   static_assert(std::is_same_v<decltype(begin(n0)), double*>);
   static_assert(std::is_same_v<decltype(end(n0)), double*>);
   static_assert(std::is_same_v<double*, decltype(n0)::pointer>);
   static_assert(noexcept(begin(n0)));
   static_assert(noexcept(end(n0)));
 
-  Kokkos::Array<double, 0> const c0{};
+  [[maybe_unused]] Kokkos::Array<double, 0> const c0{};
   static_assert(std::is_same_v<decltype(begin(c0)), double const*>);
   static_assert(std::is_same_v<decltype(end(c0)), double const*>);
   static_assert(std::is_same_v<double const*, decltype(c0)::const_pointer>);
   static_assert(noexcept(begin(c0)));
   static_assert(noexcept(end(c0)));
 
-  Kokkos::Array<double, 1> n1{};
+  [[maybe_unused]] Kokkos::Array<double, 1> n1{};
```

Comment:
What about:
```suggestion
  constexpr Kokkos::Array<double, 1> n1{};
```

(I think it's worth giving a try)

Reply:
no that would make it const

Reply:
(and fail)